cloudsync_onedrive.py

```diff
@@ -381,44 +395,74 @@ def _direct_api(self, action, path=None, *, url=None, stream=None, data=None, he
 
         return res
 
+    def _direct_api_error_trap(self, path, method="get", default=None):
+        # wrapper for _direct_api that raises only connectivity errors - other errors are caught and logged
+        try:
+            return self._direct_api(method, path)
+        except (CloudDisconnectedError, CloudTokenError, CloudTemporaryError):
+            raise
+        except Exception as e:
+            self._save_namespace_error(f"{path} failed with {repr(e)}")
+            return default
+
+    def _save_namespace_error(self, error: str):
+        _ = Drive(id="", name=error, parent=self._namespace_errors)
+        log.warning(error)
+
     def _save_drive_info(self, parent, drive_json):
-        ids = f"{parent.id}|{drive_json['id']}"
-        owner = drive_json.get("owner")
-        owner_type = list(owner)[0] if owner else ""
-        owner_id = owner[owner_type].get("id", "") if owner else ""
-        owner_name = owner[owner_type].get("displayName", "") if owner else ""
-        drive = Drive(f'{parent.name}/{drive_json.get("name", "Personal")}', ids,
-                      parent=parent,
-                      url=drive_json.get("webUrl"),
-                      owner=owner_name,
-                      owner_id=owner_id,
-                      owner_type=owner_type)
-        self.__drive_by_id[ids] = drive
+        try:
+            ids = f"{parent.id}|{drive_json['id']}"
+            owner = drive_json.get("owner")
+            owner_type = list(owner)[0] if owner else ""
+            owner_id = owner[owner_type].get("id", "") if owner else ""
+            owner_name = owner[owner_type].get("displayName", "") if owner else ""
+            drive = Drive(f'{parent.name}/{drive_json.get("name", "Personal")}', ids,
+                          parent=parent,
+                          url=drive_json.get("webUrl"),
+                          owner=owner_name,
+                          owner_id=owner_id,
+                          owner_type=owner_type)
+            self.__drive_by_id[ids] = drive
+        except Exception as e:
+            self._save_namespace_error(f"Failed to parse drive json: {drive_json} {repr(e)}")
```

Comment:
Because this is a significant part of the changes, it makes sense to get coverage on the new exception handlers